main.py

```diff
@@ -47,25 +55,39 @@ def catch_all(path):
             REQUESTS.pop(0)
 
     # Here we search the RESPONSES object to see if we have a response to return.
-    # TODO: currently this only checks the path and query args, we should add another parameter "method" so we can
-    # allow users to specify different responses based on their request method of choice.
-    potential_responses = RESPONSES.get(cr.path)
+    potential_responses = None
+
+    for pattern, responses in RESPONSES.items():
+        match = re.match(pattern, cr.path)
+        if match:
+            if potential_responses:
+                raise IndexError('Multiple path patterns match, please reconfigure responses.')
+
+            potential_responses = responses
+            # Any named groups in path will be extracted here
+            path_params = match.groupdict()
+
+    if potential_responses is None:
+        raise NotFound
+
+    for resp in potential_responses:
+        if args_match(resp, cr) and methods_match(resp, cr):
+            data = resp['response']
 
-    if potential_responses is not None:
-        for resp in potential_responses:
-            if resp['args'] == cr.args:
-                data = resp['response']
+            if isinstance(data, dict):
+                # extra brackets are to prevent format from interpreting keys as references
+                data = '{' + json.dumps(data) + '}'
 
-                if isinstance(data, dict):
-                    return jsonify(**data)
-                else:
-                    return data
+            data = data.format(**path_params)
+            return data, resp.get('status_code', 200)
 
-    return jsonify(success=True)
+    # TODO: MethodNotAllowed is raised both if methods didn't match and if no args didn't match, but ideally the second
+    # should raise a different HTTP error.
```

Comment:
Not sure, TBH. There doesn't seem to be a standard on what to do if your argument combo causes the API to return no data (404 or empty list seem to be the most common). It would honestly depend on what you are trying to mock, so it almost makes sense to make it configurable if there is a significant amount of variety out in the world.